protocol-units/settlement/mcr/contracts/script/MovementDeployer.s.sol

```diff
@@ -0,0 +1,269 @@
+    string public stakingSignature = "initialize(address)";
+    string public stlMoveSignature = "initialize(string,string,address)";
+    string public moveSignature = "initialize()";
+    address public ZERO = address(0x0);
+
+    struct Deployment {
+        address mcr;
+        address mcrAdmin;
+        address staking;
+        address stakingAdmin;
+        address move;
+        address moveAdmin;
+        address stlMove;
+        address stlMoveAdmin;
+        address timelock;
+        address multisig;
+    }
+
+    function run() external {
+        uint256 minDelay = 1 days;
+        address[] memory proposers = new address[](5);
+        address[] memory executors = new address[](1);
+        proposers[0] = helper.s2a("Andy");
+        proposers[1] = helper.s2a("Bob");
+        proposers[2] = helper.s2a("Charlie");
+        proposers[3] = helper.s2a("David");
+        proposers[4] = helper.s2a("Eve");
+        executors[0] = helper.s2a("MultisigAddress");
+        address adminAddress = ZERO;
+
+        _loadDeployments();
+
+        uint256 signer = vm.envUint("PRIVATE_KEY");
+        vm.startBroadcast(signer);
+
+        if (deployment.timelock == ZERO) {
+            timelock = new TimelockController(minDelay, proposers, executors, adminAddress);
+            deployment.timelock = address(timelock);
+        }
+        
+        deployment.moveAdmin == ZERO && deployment.move == ZERO ?
+            _deployMove() : deployment.moveAdmin != ZERO && deployment.move != ZERO ?
+                _upgradeMove() : revert("MOVE: both admin and proxy should be registered");
+
+        deployment.stakingAdmin == ZERO && deployment.staking == ZERO && deployment.move != ZERO ?
+            _deployStaking() : deployment.stakingAdmin != ZERO && deployment.staking != ZERO ?
+                _upgradeStaking() : revert("STAKING: both admin and proxy should be registered");
+
+        deployment.stlMoveAdmin == ZERO && deployment.stlMove == ZERO && deployment.move != ZERO ?
+            _deployStlMove() : deployment.stlMoveAdmin != ZERO && deployment.stlMove != ZERO ?
+                _upgradeStlMove() : revert("STL: both admin and proxy should be registered");
+
+        deployment.mcrAdmin == ZERO && deployment.mcr == ZERO && deployment.move != ZERO && deployment.staking != ZERO ?
+            _deployMCR(proposers) : deployment.mcrAdmin != ZERO && deployment.mcr != ZERO ?
+                _upgradeMCR() : revert("MCR: both admin and proxy should be registered");
+
+        vm.stopBroadcast();
+
+        if (record) {
+            _writeDeployments();
+        }
+    }
+
+    function _loadDeployments() internal{
+        // load deployments
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script/helpers/deployments.json");
+        string memory json = vm.readFile(path);
+        bytes memory rawDeploymentData = json.parseRaw(string(abi.encodePacked(".", "mainnet")));
```

Comment:
Can't be mainnet hardcoded as we will want to run the pipeline to test against Holesky.